### PR TITLE
Add Contributing guidelines

### PR DESCRIPTION
**Pull request by andrewdavidmackenzie**
Add CONTRIBUTING.md with a description of process to contribute, and guidelines for getting your contributions accepted. Plus a reference to this from the main README.md file

### CODE REVIEW

**Comment by jmprusi** [comment]
👍

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+We gladly accept contributions in the form of reproducible issues and pull requests. Please follow the guidelines here to help your work get accepted.
+
+## Reporting issues
+
+When filing a new Issue:
+
+- Please make clear in the subject what the issue is about.
+- Include all the necessary information to be able to reproduce the issues, including Operating System and version, NGINX and OpenResty version, etc.
+- Please to not include any confidential information (ids, keys, application secrets). If this is needed to reproduce and fix an issue then you maybe contacted by email.
```
Comment (davidor) [inline CONTRIBUTING.md]: typo: 'do not include...'

Comment (davidor) [inline CONTRIBUTING.md]: Also 'you may be...'

Comment (andrewdavidmackenzie) [inline CONTRIBUTING.md]: OK, fixing now...

**Comment by davidor** [comment]
There's already a `doc` dir in this repo. I think that `CONTRIBUTING.md` should be there. Other than that, :+1:

**Comment by andrewdavidmackenzie** [comment]
I thought on that, but thought maybe good to leave it more visible at top level?

WDYT?

**Comment by davidor** [comment]
The document is already linked from the readme, so I think it's already visible, but I don't have a strong preference.

Also, I noticed that there's already a `CONTRIBUTING.md` document in the `.github` dir: https://github.com/3scale/apicast/blob/v2/.github/CONTRIBUTING.md I think this can be a bit confusing.

**Comment by andrewdavidmackenzie** [comment]
I merged the content with the existing .github/CONTRIBUTING.md file and changed the link from README.md.

Now users can see these guidelines when reading README, or by following the link provided by Github when about to add an issue.
screen shot 2016-11-29 at 07 59 54

**Comment by mikz** [comment]
We probably could move the `CONTRIBUTING.md` to the repository root so it is obvious.

**Pull request merged**